float32[] ball_coordinates
float32[] container_coordinates